File: registers/out_register/OUTRegister.py
from m5.params import *
from m5.SimObject import SimObject

class OUTRegister(SimObject):
    type = 'OUTRegister'
    cxx_header = "CAProject/registers/out_register/out_register.hh"
    cxx_class = "gem5::OUTRegister"

    out = Param.UInt8(0, "Output stored in the register")

File: registers/out_register/out_register.cc
#include "CAProject/registers/out_register/out_register.hh"

namespace gem5
{

OUTRegister::OUTRegister(const OUTRegisterParams &params) : 
    SimObject(params),
    out(params.out)
{
}

void
OUTRegister::setValue(uint8_t val)
{
    out = val;
}

uint8_t
OUTRegister::getValue()
{
    return out;
}

} // namespace gem5
